*** boinc-trunk-rev25985/lib/synch.cpp	2012-08-05 10:13:58.082035988 +0200
--- boinc-trunk-rev25985_android/lib/synch.cpp	2012-08-05 09:55:52.354834255 +0200
***************
*** 18,26 ****
  
  // interfaces for accessing sempahores
  
! #include "synch.h"
  
! #ifndef ANDROID //disabled on Android
  
  #include "config.h"
  #include <cstdio>
--- 18,26 ----
  
  // interfaces for accessing sempahores
  
! #ifdef HAVE_SYS_SEM_H
  
! #include "synch.h"
  
  #include "config.h"
  #include <cstdio>
***************
*** 114,117 ****
      return 0;
  }
  
! #endif //ANDROID
--- 114,117 ----
      return 0;
  }
  
! #endif //HAVE_SYS_SEM_H
